src/Features/OnboardingTasks/Tasks/Purchase.php

```diff
@@ -149,8 +149,19 @@ public static function get_products() {
 		$installed     = PluginsHelper::get_installed_plugin_slugs();
 		$product_types = isset( $profiler_data['product_types'] ) ? $profiler_data['product_types'] : array();
 		$product_data  = Onboarding::get_product_data( Onboarding::get_allowed_product_types() );
-		$purchaseable  = array();
-		$remaining     = array();
+		$themes        = Onboarding::get_themes();
+		$theme_key     = array_search( $profiler_data['theme'], array_column( $themes, 'slug' ), true );
+		if ( false !== $theme_key && isset( $themes[ $theme_key ]['id'] ) && isset( $themes[ $theme_key ]['price'] ) && ! $themes[ $theme_key ]['is_installed'] ) {
```

Comment:
Not a deal-breaker, but you could consider breaking this down a bit to make it easier to understand

Reply:
You could also declare `$theme = $themes[ $theme_key ];` in the `if` statement to make it a bit simpler